packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -215,9 +215,16 @@ class OverlayState extends State<Overlay> {
   }
 
   void _remove(OverlayEntry entry) {
-    setState(() {
-      _entries.remove(entry);
+    assert(() {
+      if (!mounted)
+        assert(_entries.remove(entry));
```

Comment:
Maybe it's worth doing a little extra work in release mode to avoid having different behavior? This doesn't seem performance critical.